### DISCUSSION TITLE
[@types/node] undici-types errors with PNPM and Angular 17 project

### DISCUSSION THREAD

**Discussion by mikeandtherest** [DefinitelyTyped/DefinitelyTyped#68146] [Issues with a @types package]
Node: 20.10.0 LTS
PNPM: 8.14.0
@types/node: 20.10.7
Angular 17.0.9 

Hi there.

I'm getting the errors from the screenshot below, after upgrading a project, which uses PNPM, from Angular v16 to v17.

The workaround is to do `pnpm i -D undici-types` and manually install it to the devDependencies.
However, by doing that I'm duplicating the package install, since it is brought already by @types/node installation. Just that it's placed under "node_modules\.pnpm\undici-types@6.3.0" path, not on the root of node_modules. It seems like a module resolution issue.

Btw, under the "node_modules\.pnpm\" path, I have both an undici-types@6.3.0 and an undici-types@5.26.5 folder, I don't know what package is installing the latter. Could this be a/the problem or it's fine?

I'm good for now with the workaround, but it would be better to not duplicate the package inside node modules. Thanks!

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/16487780/c14d1710-3933-4fc3-9664-b0b146e20ab5)

[a comment by an automated account is not shown]

**Comment by mikeandtherest** [comment]
It's still the same after one week. Is anyone able to help with this please?

**Reply by Semigradsky** [reply]
It looks like pnpm issue, isn't it?

**Reply by mikeandtherest** [reply]
It might be the case, but it would be weird. Because pnpm didn't change during the Angular v16 to v17 upgrade, only @types/node changed to a higher version. Which makes me think that it must be caused by something inside @types/node.

**Reply by Semigradsky** [reply]
This issue sounds similar, maybe it will help you - https://github.com/pnpm/pnpm/issues/4920
